app/src/main/java/com/github/epfl/meili/util/PoiServiceViewModel.kt

```diff
@@ -0,0 +1,72 @@
+package com.github.epfl.meili.util
+
+import android.location.Location
+import android.location.LocationListener
+import android.os.Bundle
+import android.widget.Toast
+import com.github.epfl.meili.MainApplication
+import com.github.epfl.meili.models.PointOfInterest
+import com.github.epfl.meili.poi.PoiService
+import com.google.android.gms.maps.model.LatLng
+
+interface PoiServiceViewModel : LocationListener {
+    companion object {
+        const val MAX_NUM_REQUESTS = 2
+    }
+
+    var poiService: PoiService?
+    var nbCurrentRequests: Int
+    var lastUserLocation: LatLng?
+
+    fun initPoiService(service: PoiService) {
+        this.poiService = service
+        if (lastUserLocation != null) {
+            requestPois()
+        }
+    }
+
+    private fun requestPois() {
+        poiService!!.requestPois(
+            lastUserLocation!!,
+            { poiList -> onSuccessPoiReceived(poiList) },
+            { error -> onError(error) })
+    }
+
+    fun onSuccessPoiReceived(poiList: List<PointOfInterest>) {
+        nbCurrentRequests = 0
+    }
```

Comment:
Was this function originally just setting nbCurrentRequests to 0? If I recall correctly request POIs should accept an onSuccess callback, am I wrong? Not really sure how it is working right now if once you call requestPois there is nothing returned.

Reply:
Okay so I see that you are overriding the function in MapActivity. Wouldn't it be possible to put the observables and the logic behind it here?

Reply:
So what I was going for here is to have the logic of retrying in the PoiServiceViewModel so that the view models which inherit it don't have to worry about this.

Reply:
What is not ideal in my implementation is that the inheriting view models still need to override the `nbCurrentRequests` variable and set it to 0, but I didn't find a way around this since `PoiServiceViewModel` is an interface.

Reply:
So we finished to discuss this offline, and we decided that it makes more sense to do it as @aman3014 is already doing it right now.